TFUtil.py

```diff
@@ -1028,10 +1028,11 @@ def get_placeholder_flattened(self, keep_dims=False):
       ndim -= 1
     if len(dyn_axes) > 1:
       shape = tf.shape(x)
+      shape_as_list = x._shape_as_list()
```

Comment:
You are calling a hidden method. You should not do that. Also, why do you need that?